### PR TITLE
Create `actors_measures` table and `actor_measures` API

### PR DESCRIPTION
**Pull request by parndt**
Implements #20  

We need a many to many join table describing an "active" link between an
actor and an activity/measure `actors`, introduced for #18, and
`measures`, from impact (and updated in #31).

##### Permissions by role

- public: none
- registered, no role: none
- analyst: R
- manager: CRUD
- admin: CRUD

##### Rules

- only active actors where allowed for its type (`actor.actortype.is_active = true`)

This also fixes a policy error where we were showing draft records to
anyone, as long as the table had a `draft` column. Now, it will only do
this for people who have an `analyst` role.

### CODE REVIEW

**Comment by tmfrnz** [comment]
@parndt, note that analysts should not be able to see draft content, but only content where draft = false
https://github.com/dumparkltd/marine-defrag-server/issues/2#issuecomment-946700727

File: config/routes.rb [hunk 1]
```diff
@@ -8,8 +8,9 @@
   end
   get "static_pages/home"
 
-  resources :actors
   resources :actor_types, only: [:index, :show]
+  resources :actors
+  resources :actors_measures, only: [:index, :show]
```
Comment (tmfrnz) [inline config/routes.rb:13]: to be consistent with other routes it should be `actor_measures`

File: app/policies/application_policy.rb [hunk 2]
```diff
@@ -29,7 +29,13 @@ class Scope
     attr_reader :user, :scope
     def resolve
       return scope.all if @user.role?("admin") || @user.role?("manager")
-      return scope.where(draft: false) if scope.column_names.include?("draft")
+      if @user.role?("analyst")
```
Comment (tmfrnz) [inline app/policies/application_policy.rb:32]: looks like this does the right thing (https://github.com/dumparkltd/marine-defrag-server/pull/56#issuecomment-957252451):

if tables can have draft content, it is only visible to analysts when it is not draft.
if tables can NOT have draft content, analyst can see it

Comment (parndt) [reply app/policies/application_policy.rb:32]: Awesome 😄

**Comment by parndt** [comment]
Just implementing this rule:

- only active actors where allowed for its type (`actor.actortype.is_active == true`)

**Comment by parndt** [comment]
> Just implementing this rule:
> 
>     * only active actors where allowed for its type (`actor.actortype.is_active == true`)

done ✅

**Comment by parndt** [comment]
This contains code from #61 so needs that merged first

**Comment by parndt** [comment]
> This contains code from #61 so needs that merged first

✅

File: db/migrate/20211103073220_create_actors_measures.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+class CreateActorsMeasures < ActiveRecord::Migration[6.1]
+  def change
+    create_table :actors_measures do |t|
```
Comment (tmfrnz) [inline db/migrate/20211103073220_create_actors_measures.rb:3]: should be called actor_measures for consistency

File: app/models/actor_measure.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class ActorMeasure < VersionedRecord
+  self.table_name = :actors_measures
```
Comment (tmfrnz) [inline app/models/actor_measure.rb:2]: should be called actor_measures

**Review by tmfrnz — APPROVED** [review]

**Pull request merged**